File: android/src/main/java/com/mrousavy/camera/parsers/Torch.kt
package com.mrousavy.camera.parsers

enum class Torch(override val unionValue: String): JSUnionValue {
  OFF("off"),
  ON("on");

  companion object: JSUnionValue.Companion<Torch> {
    override fun fromUnionValue(unionValue: String?): Torch {
      return when (unionValue) {
        "off" -> OFF
        "on" -> ON
        else -> OFF
      }
    }
  }
}

File: ios/Core/Types/Torch.swift
//
//  Torch.swift
//  VisionCamera
//
//  Created by Marc Rousavy on 11.10.23.
//  Copyright © 2023 mrousavy. All rights reserved.
//

import AVFoundation
import Foundation

/**
 A Torch used for permanent flash.
 */
@frozen
enum Torch: String, JSUnionValue {
  /**
   Torch (flash unit) is always off.
   */
  case off
  /**
   Torch (flash unit) is always on.
   */
  case on

  init(jsValue: String) throws {
    if let parsed = Torch(rawValue: jsValue) {
      self = parsed
    } else {
      throw CameraError.parameter(.invalid(unionName: "torch", receivedValue: jsValue))
    }
  }

  var jsValue: String {
    return rawValue
  }

  func toTorchMode() -> AVCaptureDevice.TorchMode {
    switch self {
    case .on:
      return .on
    case .off:
      return .off
    }
  }
}
